src/test/kotlin/fi/metatavu/pakkasmarja/services/erp/test/functional/tests/BusinessPartnersResourceTest.kt

```diff
@@ -31,61 +32,51 @@ class BusinessPartnersResourceTest: AbstractResourceTest() {
-
-            assertEquals("MetaLab", partner.companyName)
-            assertEquals("0000000", partner.federalTaxId)
-
-            assertEquals(getTestValidationDate(), partner.updated)
-
-            val bankAccounts = partner.bankAccounts
-            assertNotNull(bankAccounts)
-            assertEquals(1, bankAccounts!!.size)
-
-            val bankAccount = bankAccounts[0]
-            assertEquals("FI61000000000", bankAccount.IBAN)
-            assertEquals("SBANFIHH", bankAccount.BIC)
-        }
-    }
+            SapMock().use { sapMock ->
+                sapMock.mockBusinessPartners("1", "2", "3")
 
-    /**
-     * Tests listing business partners with a null access token
-     */
-    @Test
-    fun testListBusinessPartnersNullAccessToken() {
-        createTestBuilder().use {
-            it.nullAccess.businessPartners.assertListFailStatus(expectedStatus = 401, updatedAfter = getTestDate())
-        }
-    }
+                val businessPartners = it.manager.businessPartners.listBusinessPartners(updatedAfter = getTestDate())
+                assertEquals(3, businessPartners.size)
+
+                val partner = businessPartners.find{ sapBusinessPartner -> sapBusinessPartner.vatLiable == SapBusinessPartner.VatLiable.EU }!!
+                assertEquals(3, partner.code)
+                assertEquals("jorma@example.com", partner.email)
+
+                val phoneNumbers = partner.phoneNumbers
+                assertNotNull(phoneNumbers)
+                assertEquals(2, phoneNumbers!!.size)
+                assertEquals("0440120122", phoneNumbers[0])
+                assertEquals("0440120123", phoneNumbers[1])
+
+                val addresses = partner.addresses
+                assertNotNull(addresses)
+                assertEquals(1, addresses!!.size)
+                val address = addresses[0]
+                assertEquals(SapAddressType.DELIVERY, address.type)
+                assertEquals("Home", address.name)
+                assertEquals("Mikkeli", address.city)
+                assertEquals("Hallituskatu 7", address.streetAddress)
+                assertEquals("50100", address.postalCode)
+
+                assertEquals("MetaLab", partner.companyName)
+                assertEquals("0000000", partner.federalTaxId)
+
+                assertEquals(getTestValidationDate(), partner.updated)
+
+                val bankAccounts = partner.bankAccounts
+                assertNotNull(bankAccounts)
+                assertEquals(1, bankAccounts!!.size)
+
+                val bankAccount = bankAccounts[0]
+                assertEquals("FI61000000000", bankAccount.IBAN)
+                assertEquals("SBANFIHH", bankAccount.BIC)
+
+                it.invalidAccess.businessPartners.assertListFailStatus(expectedStatus = 401, updatedAfter = getTestDate())
+                it.nullAccess.businessPartners.assertListFailStatus(expectedStatus = 401, updatedAfter = getTestDate())
+
+                sapMock.close()
```

Comment:
SapMock is AutoCloseable, no need to close it manually

Reply:
Fixed